### PR TITLE
feat: track contributors in workspaces

### LINKED ISSUE #2643: Feature: Workspaces: Import individual contributors to track
As a user of workspaces, I want to be able to compile a list of contributors to track by searching for contributors on an individual basis.

CleanShot 2024-02-12 at 17 22 53@2x

image

### PR DESCRIPTION
**Pull request by zeucapua**
## Description

Implements tracking contributors in Workspaces. There are two parts for basic functionality:

- Add contributors when creating a workspace under `/workspaces/new`
  - This will done in a future PR
- [X] Updating contributors in the Settings (individual search to add, table, updating)
  - [X] `TrackedContributorsTable` 
    - [x] initial fetch
    - [X] delete contributor (complete with `updateWorkspace()`)
  - [X] `TrackedContributorsModal` (add to Table on complete)
  - [X] `SearchByContributorsStep` 
    - add and filter contributors, uses `/v2/users/search?username` endpoint
    - other steps to be implemented in future PRs
  - [x] `updateWorkspace()` in `/workspaces/[id]/settings`
- [x] Fetching data to display under `/workspaces/[id]/contributors`
  - Text only, graphs will be implemented in a future PRs

## What type of PR is this? (check all applicable)

- [X] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Closes #2643
Relates to #2578 

## Mobile & Desktop Screenshots/Recordings

https://github.com/open-sauced/app/assets/20603494/59712f53-983f-4b03-b248-a8c1fce89aec

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Comment by zeucapua** [comment]
Redeploying on Netlify completes the build fine, I don't know how to re-run the deploy via Github to clear the Netlify checks.

**Review by bdougie — APPROVED** [review]
I did a bit of QA on this and LGTM

**Review by isabensusan — APPROVED** [review]
working great for me!

**Review by nickytonline — CHANGES_REQUESTED** [review]
This is working great @zeucapua. A few change requests.

**Review by jpmcb — APPROVED** [review]

File: lib/utils/workspace-utils.ts [hunk 1]
```diff
@@ -57,12 +60,18 @@ export async function saveWorkspace({
...
-    updateWorkspace,
-    updateWorkspaceRepos,
-  ]);
+  const updateWorkspaceContributors = await fetchApiData<any[]>({
+    path: `workspaces/${workspaceId}/contributors`,
+    method: "POST",
+    body: { contributors },
+    bearerToken: sessionToken,
+    pathValidator: () => true,
+  });
```
[the remark this answers is not in the document]
Comment (zeucapua) [reply lib/utils/workspace-utils.ts:69]: 🫡

**Review by nickytonline — APPROVED** [review]

**Pull request merged**